app/controllers/api/v1/applications_controller.rb

```diff
@@ -4,15 +4,35 @@ class ApplicationsController < ApplicationController
       include Api::V1::Mixins::DestroyMixin
       include Api::V1::Mixins::IndexMixin
       include Api::V1::Mixins::ShowMixin
-      include Api::V1::Mixins::UpdateMixin
 
       def create
         application = Application.new(params_for_create).tap { |app| authorize(app) }
         application.save!
 
-        raise_event("#{model}.create", application.as_json)
+        # we do not want to raise the create event since the application has
+        # not been processed by the superkey worker.
+        raise_event("Application.create", application.as_json) unless application.source.super_key?
```

Comment:
I'm totally outside this codebase, but can any of the `Event` concern methods be used here? `app/models/concerns/event_concern.rb` so we could hide some of the conditional logic within a `raise_event` method?

Reply:
I really like the look of #344, so I'll wait on that one and refactor this to use the `raise_event_if` pattern.